src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/softdelete/SoftDeleteService.kt

```diff
@@ -159,4 +160,11 @@ class SoftDeleteService(
     val batchSize: Int,
     val lastIdProcessed: Long?,
   )
+
+  private fun <R> runInTransaction(block: () -> R) =
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**praise:** ‏Magic!